js/backup.js

```diff
@@ -0,0 +1,659 @@
+;(function () {
+  'use strict';
+  window.Whisper = window.Whisper || {};
+
+  function stringToBlob(string) {
+    var buffer = dcodeIO.ByteBuffer.wrap(string).toArrayBuffer();
+    return new Blob([buffer]);
+  }
+
+  function stringify(object) {
+    for (var key in object) {
+      var val = object[key];
+      if (val instanceof ArrayBuffer) {
+        object[key] = {
+          type: 'ArrayBuffer',
+          encoding: 'base64',
+          data: dcodeIO.ByteBuffer.wrap(val).toString('base64')
+        };
+      } else if (val instanceof Object) {
+        object[key] = stringify(val);
+      }
```

Comment:
No, this method is a recursive check for `ArrayBuffer`, that's it. If it's not an object (recurse) or `ArrayBuffer` (transform) we don't care about it.